recipes/imagl/all/conanfile.py

```diff
@@ -44,21 +52,45 @@ def source(self):
     def validate(self):
         if self.settings.compiler.cppstd:
             tools.check_min_cppstd(self, 20)
+
+        def lazy_lt_semver(v1, v2):
+            lv1 = [int(v) for v in v1.split(".")]
+            lv2 = [int(v) for v in v2.split(".")]
+            min_length = min(len(lv1), len(lv2))
+            return lv1[:min_length] < lv2[:min_length]
+
         minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        compiler_version = str(self.settings.compiler.version)
         if minimum_version:
-            if tools.Version(self.settings.compiler.version) < minimum_version:
-                raise ConanInvalidConfiguration("imagl requires C++20, which your compiler does not fully support.")
+            if lazy_lt_semver(compiler_version, minimum_version): 
+                raise ConanInvalidConfiguration("imagl requires some C++20 features, which your compiler does not support.")
+            #Special case for Visual Studio for which conan doesn't known its minor version
```

Comment:
What I understand in Visual Studio versioning in conan, it's that it can only be 8, 9, 10, 12, 14, 15 or 16. These values come from conan profile. So I can not test for "16.2"
So, what I do here is to find latest Visual Studio installation, set its environment and get VCToolsVersion variable which is the toolset version with its minor value (14.20 to 14.28)

Reply:
OK... It doesn't work on your jenkins config... There is no %ProgramFiles(x86)% environement variable ?? Is it a 32-bit Windows ??
I'm totally stuck...

Reply:
Oups, every thing is in conan tools package!! I stop for tonight.